### DISCUSSION TITLE
Allow pre-processor directives in XML comments

### DISCUSSION THREAD

**Discussion by TaffarelJr** [dotnet/csharplang#7567] [Language Ideas]
Currently the lexical specification does not allow elements like pre-processor directives to be interspersed with XML documentation. This means we get compiler errors when trying to do something like this:

![image](https://github.com/dotnet/csharplang/assets/5290662/2e7663b5-c452-41d7-ab58-050249aa79c2)
> Originally filed in [dotnet/roslyn #70057](https://github.com/dotnet/roslyn/issues/70057)

I believe it is a valid scenario to want to have specific XML documenation for different platforms when multi-targeting, _without_ having to duplicate entire blocks of XML comments.

What is the feasibility of such a feature being supported?

**Comment by CyrusNajmabadi** [comment]
Ok a technical level it would be extremely challenging. These are different trivia structures, so the content would be be split.  The XML system would then have to be rewritten to not have everything be contiguous.

**Comment by jnm2** [comment]
Existing discussion: https://github.com/dotnet/csharplang/discussions/295
